compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/BridgeToAIRPass.cpp

```diff
@@ -0,0 +1,96 @@
+// Copyright 2023 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include "iree-amd-aie/Transforms/PassDetail.h"
+#include "iree-amd-aie/Transforms/Passes.h"
+#include "mlir/Dialect/Arith/Utils/Utils.h"
+#include "mlir/Dialect/MemRef/IR/MemRef.h"
+#include "mlir/Dialect/SCF/IR/SCF.h"
+#include "mlir/Dialect/Utils/StaticValueUtils.h"
+#include "mlir/IR/PatternMatch.h"
+#include "mlir/Transforms/GreedyPatternRewriteDriver.h"
+
+namespace mlir::iree_compiler::AMDAIE {
+
+namespace {
+
+/// Pattern to rewriter scf.forall -> scf.parallel after bufferization.
+class SCFForAllToParallelOp : public OpRewritePattern<scf::ForallOp> {
+  using OpRewritePattern<scf::ForallOp>::OpRewritePattern;
+
```

Comment:
This would seem like a something that should either be handled by the existing mlir-aie code, or should be a general canonicalization pattern?

Reply:
Cant be a general canonicalization pattern. `scf.parallel` is kind of legacy (and actually IMO `scf.forall` and this are redundant, one of these need to go). It could be in AIE.